src/lib/dnssd/minimal_mdns/ResponseSender.h

```diff
@@ -47,6 +47,24 @@ namespace Minimal {
 
 namespace Internal {
 
+// Flags for keeping track of items having been sent as DNSSD responses
+//
+// We rely on knowing Matter DNSSD only sends the same set of data
+// for some instances like A/AAAA always being the same.
```

Comment:
Is that true, given DHCP?

Reply:
It is true for the given MinMDNS implementation. A/AAAA reporting is a `loop(nextIP) { encode(nextIP)}` so for a given query we always return the same set of data.

It may change over time, however within a single reply we expect the addresses to be the same.